helix-rest/src/main/java/org/apache/helix/rest/client/CustomRestClientImpl.java

```diff
@@ -52,16 +56,26 @@
   private static final String ACCEPT_CONTENT_TYPE = "application/json";
   private static final ObjectMapper OBJECT_MAPPER = new ObjectMapper();
 
-  private HttpClient _httpClient;
+  private final HttpClient _httpClient;
+  private final Semaphore _semaphore;
```

Comment:
Could we add a block comment here on why it's necessary to have a semaphore here (just how you did for cachedResults)?